script/LocalDeploy.s.sol

```diff
@@ -24,119 +25,212 @@ import {OrigamiTimelockController} from "src/OrigamiTimelockController.sol";
-        DiamondCutFacet diamondCutFacet = new DiamondCutFacet();
-        console2.log("DiamondCutFacet: ", address(diamondCutFacet));
-        facetAddresses["DiamondCutFacet"] = address(diamondCutFacet);
+        if (isUndeployed("DiamondCutFacet")) {
+            DiamondCutFacet diamondCutFacet = new DiamondCutFacet();
+            addresses["DiamondCutFacet"] = address(diamondCutFacet);
+        }
 
-        DiamondLoupeFacet diamondLoupeFacet = new DiamondLoupeFacet();
-        console2.log("DiamondLoupeFacet: ", address(diamondLoupeFacet));
-        facetAddresses["DiamondLoupeFacet"] = address(diamondLoupeFacet);
+        if (isUndeployed("DiamondLoupeFacet")) {
+            DiamondLoupeFacet diamondLoupeFacet = new DiamondLoupeFacet();
+            addresses["DiamondLoupeFacet"] = address(diamondLoupeFacet);
+        }
 
-        OwnershipFacet ownershipFacet = new OwnershipFacet();
-        console2.log("OwnershipFacet: ", address(ownershipFacet));
-        facetAddresses["OwnershipFacet"] = address(ownershipFacet);
+        if (isUndeployed("OwnershipFacet")) {
+            OwnershipFacet ownershipFacet = new OwnershipFacet();
+            addresses["OwnershipFacet"] = address(ownershipFacet);
+        }
 
-        GovernorCoreFacet governorCoreFacet = new GovernorCoreFacet();
-        console2.log("GovernorCoreFacet: ", address(governorCoreFacet));
-        facetAddresses["GovernorCoreFacet"] = address(governorCoreFacet);
+        if (isUndeployed("GovernorCoreFacet")) {
+            GovernorCoreFacet governorCoreFacet = new GovernorCoreFacet();
+            addresses["GovernorCoreFacet"] = address(governorCoreFacet);
+        }
 
-        GovernorSettingsFacet governorSettingsFacet = new GovernorSettingsFacet();
-        console2.log("GovernorSettingsFacet: ", address(governorSettingsFacet));
-        facetAddresses["GovernorSettingsFacet"] = address(governorSettingsFacet);
+        if (isUndeployed("GovernorSettingsFacet")) {
+            GovernorSettingsFacet governorSettingsFacet = new GovernorSettingsFacet();
+            addresses["GovernorSettingsFacet"] = address(governorSettingsFacet);
+        }
 
-        GovernorTimelockControlFacet governorTimelockControlFacet = new GovernorTimelockControlFacet();
-        console2.log("GovernorTimelockControlFacet: ", address(governorTimelockControlFacet));
-        facetAddresses["GovernorTimelockControlFacet"] = address(governorTimelockControlFacet);
+        if (isUndeployed("GovernorTimelockControlFacet")) {
+            GovernorTimelockControlFacet governorTimelockControlFacet = new GovernorTimelockControlFacet();
+            addresses["GovernorTimelockControlFacet"] = address(governorTimelockControlFacet);
+        }
     }
 
-    function deployProxies() public {
-        ProxyAdmin admin = new ProxyAdmin();
-        console2.log("ProxyAdmin: ", address(admin));
-        proxyAddresses["ProxyAdmin"] = address(admin);
+    function emitRunFile() public {
```

Comment:
Does this only emit one file or does it emit one file per dao you make. It looks like this would only work with one additional local dao but seems like a couple of small tweaks might let us be able to spin up as many new daos as we want.

Reply:
Ah. thanks for clarifying. That's great!